fdb-extensions/src/main/java/com/apple/foundationdb/tuple/TupleHelpers.java

```diff
@@ -93,6 +94,34 @@ public static boolean equals(@Nullable Tuple t1, @Nullable Tuple t2) {
         }
     }
 
+    /**
+     * Negate a number used as an element of a {@link Tuple}.
+     * @param number the number to be negated
+     * @return a negated number suitable for use in a new {@link Tuple}.
+     */
+    public static Number negate(@Nonnull Number number) {
+        if (number instanceof Long || number instanceof Integer || number instanceof Short || number instanceof Byte) {
+            long l = number.longValue();
+            if (l == Long.MIN_VALUE) {  // The only long whose negation is not a long.
+                return BigInteger.valueOf(l).negate();
+            } else {
+                return - l;
+            }
+        }
+        if (number instanceof BigInteger) {
+            BigInteger n = ((BigInteger)number).negate();
+            if (n.equals(BigInteger.valueOf(Long.MIN_VALUE))) {
+                return Long.MIN_VALUE;  // The normal form of this number.
```

Comment:
Hm, is this so that `negate(negate(Long.MIN_VALUE)) == Long.MIN_VALUE`? Is this a property we need?

Reply:
I guess not, provided you immediately stick the result into a `Tuple`. It will be denormalized, with a `BigInteger` where a `Long` would normally be. But the comparison, etc. operations handle that case, as they do an `Integer` where a `Long` would be.

Does it seem superfluous?

Reply:
Hm, I suppose it's probably fine. I suspect that the "right" thing would be to make sure that all of our comparison operations handle comparing between types correctly, but perhaps it is safer to just convert to a `Long`.